webapp-client/src/component/ContentPane.js

```diff
@@ -0,0 +1,78 @@
+import '../stylesheets/ContentPane.css';
+import errorify from '../errorify.js';
+import Item from './Item';
+import PropTypes from 'prop-types';
+import React from 'react';
+
+export default class ContentPane extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      items: [],
+    };
+    errorify(this);
+  }
+
+  componentDidMount() {
+    this.setItems(this.props);
+  }
+
+  componentDidUpdate(prevProps) {
+    if (this.props !== prevProps) {
+      this.setItems(this.props);
+    }
+  }
+
+  async setItems(props) {
+    const { categoryId, defaultCategory, userId } = props;
+    let url;
+    let response;
+    let items;
+    try {
+      if (categoryId === defaultCategory) {
+        url = process.env.REACT_APP_API_URL + '/item/getItems/' + userId;
+        response = await fetch(url);
+        items = await response.json();
+      } else {
+        url =
+          process.env.REACT_APP_API_URL +
+          '/category/getCategoryItems/' +
+          categoryId;
+        response = await fetch(url);
+        const categoryObject = await response.json();
+        items = categoryObject.items;
+      }
+      if (response.status !== 200) {
+        throw new Error(response.statusMessage);
```

Comment:
Is this message something you want to display to the user? IIRC the server is just passing back the default error message from Mongo.

Reply:
Does this error display to the user? I thought it would get caught and trigger the generic error display

Reply:
Oops you're right. Mixed this up with a comment I left on the previous PR!